nucypher/config/base.py

```diff
@@ -233,53 +233,62 @@ def _ensure_config_root_exists(self) -> None:
+        if not self.config_root.exists():
             try:
-                os.mkdir(self.config_root, mode=0o755)
+                self.config_root.mkdir(mode=0o755)
             except FileNotFoundError:
-                os.makedirs(self.config_root, mode=0o755)
+                self.config_root.mkdir(parents=True, mode=0o755)
 
     @classmethod
-    def peek(cls, filepath: str, field: str) -> Union[str, None]:
+    def peek(cls, filepath: Path, field: str) -> Union[str, None]:
         payload = cls._read_configuration_file(filepath=filepath)
         try:
             result = payload[field]
         except KeyError:
             raise cls.ConfigurationError(f"Cannot peek; No such configuration field '{field}', options are {list(payload.keys())}")
         return result
 
-    def to_configuration_file(self, filepath: str = None, modifier: str = None, override: bool = False) -> str:
+    def to_configuration_file(self, filepath: Path = None, modifier: str = None, override: bool = False) -> Path:
         filepath = self.generate_filepath(filepath=filepath, modifier=modifier, override=override)
         self._ensure_config_root_exists()
         filepath = self._write_configuration_file(filepath=filepath, override=override)
         return filepath
 
     @classmethod
-    def from_configuration_file(cls, filepath: str = None, **overrides) -> 'BaseConfiguration':
+    def from_configuration_file(cls, filepath: Path = None, **overrides) -> 'BaseConfiguration':
         filepath = filepath or cls.default_filepath()
         payload = cls._read_configuration_file(filepath=filepath)
         instance = cls(filepath=filepath, **payload, **overrides)
         return instance
 
     @classmethod
-    def _read_configuration_file(cls, filepath: str) -> dict:
+    def _read_configuration_file(cls, filepath: Path) -> dict:
         """Reads `filepath` and returns the deserialized JSON payload dict."""
         with open(filepath, 'r') as file:
             raw_contents = file.read()
-            payload = cls.deserialize(raw_contents, payload_label=filepath)
+            payload = cls.deserialize(raw_contents, payload_label=str(filepath))
         return payload
 
-    def _write_configuration_file(self, filepath: str, override: bool = False) -> str:
+    def _write_configuration_file(self, filepath: Path, override: bool = False) -> Path:
         """Writes to `filepath` and returns the written filepath.  Raises `FileExistsError` if the file exists."""
-        if os.path.exists(str(filepath)) and not override:
+        if filepath.exists() and not override:
             raise FileExistsError(f"{filepath} exists and no filename modifier supplied.")
         with open(filepath, 'w') as file:
             file.write(self.serialize())
         return filepath
 
     def serialize(self, serializer=json.dumps) -> str:
         """Returns the JSON serialized output of `static_payload`"""
+        def _stringify_paths(d: dict):
```

Comment:
In general it feels like it would be a good application for `marshmallow` (with `Path` being a field, so it will be serialized/deserialized appropriately), but perhaps it's not for this PR.